strings/base_meta.h

```diff
@@ -117,24 +117,23 @@ namespace winrt::impl
         static constexpr auto data{ category_signature<category_t<T>, T>::data };
     };
 
+#if defined(__clang__)
     template <typename T>
     struct classic_com_guid
     {
-#if defined(__clang__)
 #if __has_declspec_attribute(uuid) && defined(WINRT_IMPL_IUNKNOWN_DEFINED)
         static constexpr guid value{ __uuidof(T) };
 #else
         static_assert(std::is_void_v<T> /* dependent_false */, "To use classic COM interfaces, you must compile with -fms-extensions and include <unknwn.h> before including C++/WinRT headers.");
-#endif
-#elif defined(_MSC_VER)
-        static constexpr guid value{ __uuidof(T) };
-#else
-        static_assert(std::is_void_v<T> /* dependent_false */, "Classic COM interfaces are not supported on this compiler");
 #endif
     };
 
     template <typename T>
     inline constexpr guid guid_v = classic_com_guid<T>::value;
+#else
+    template <typename T>
+    inline constexpr guid guid_v{ __uuidof(T) };
```

Comment:
Don't we want to keep the `#elif defined(_MSC_VER)` and `#else static_assert("Sorry, your compiler doesn't support classic COM interfaces.")` parts?

Reply:
So something like this?

```cpp
#if defined(__clang__)
    template <typename T>
    struct classic_com_guid
    {
#if __has_declspec_attribute(uuid) && defined(WINRT_IMPL_IUNKNOWN_DEFINED)
        static constexpr guid value{ __uuidof(T) };
#else
        static_assert(std::is_void_v<T> /* dependent_false */, "To use classic COM interfaces, you must compile with -fms-extensions and include <unknwn.h> before including C++/WinRT headers.");
#endif
    };
#elif !defined(_MSC_VER)
    template <typename T>
    struct classic_com_guid
    {
        static_assert(std::is_void_v<T> /* dependent_false */, "Sorry, your compiler doesn't support classic COM interfaces.");
    };
#endif

#if defined(__clang__) || !defined(_MSC_VER)
    template <typename T>
    inline constexpr guid guid_v = classic_com_guid<T>::value;
#else
    template <typename T>
    inline constexpr guid guid_v{ __uuidof(T) };
#endif
```

Reply:
Why do we want to proactively prohibit unknown compilers that might work?

Reply:
I initially added it as a nice error message to replace a less nicer "value: undefined identifier", but with my current commit you should get something in the lines of not recognizing __uuidof, which is friendlier.

Reply:
I guess that's okay. Hopefully gcc folks will know what it means when they get "undeclared identifier `__uuidof`."